### DISCUSSION TITLE
Keep colors on flux diff output

### DISCUSSION THREAD

**Discussion by akselleirv** [fluxcd/flux2#2783] [Q&A] [answered]
Hello all!

I want to add the output of `flux diff` in the PR to make it easier to review changes. My idea is to run the flux diff in a pipeline job and capture the output. Then post the output as a comment to the PR.

However, when the command is not executed from the terminal then the colors are removed. 

Any ideas of how to solve this?

**Comment by kingdonb** [comment, marked as answer]
I can suggest using interactive terminal config as a guess, (does Flux change its output based on whether or not it is run from an interactive terminal?)

It looks like color in the terminal is offloaded to a few libraries, one is called "bunt" and the other is "go-colorful"

It's not clear to me if they have this behavior, but since it is a common behavior at a guess I think they do.

So the output may be changed based on whether an interactive terminal is used or not... getting the CI job to run with an interactive TTY may be another different matter. Depending on how the job is run, you may or may not have access to these features. (What CI is it?)

You could also try to add the color back using some markup/markdown formatting wrapped around the output:

```diff

        "name": "vscode-gitops-tools",
-       "version": "0.19.2",
+       "version": "0.19.3-pre.1",
        "lockfileVersion": 2,
        "requires": true,
        "packages": {
                "": {
                        "name": "vscode-gitops-tools",
-                       "version": "0.19.2",
+                       "version": "0.19.3-pre.1",
                        "license": "MPL-2.0",
                        "dependencies": {
                                "@kubernetes/client-node": "^0.16.2",
```

I'm not sure how this will look for `flux diff` output but it might really be enough, as long as you can serve up git-flavored markdown in your CI solution's reporting output.

**Reply by akselleirv** [reply]
I need to pipe the output and turns off the color. I'm going to try the markdown approach, thanks for the help!

**Comment by andloh** [comment]
Old post, but it can be solved with using `unbuffer`.

I do not know if a CI will preserve it

E.g 

```bash
diff="$( unbuffer flux diff --kube-api-qps=100 kustomization $name --path $KUSTOMIZE_PATH --progress-bar=false)"
echo "$diff"
```
